javascript/javascript3/week3/main.js

```diff
@@ -0,0 +1,31 @@
+
+const redBox = document.querySelector('ul.marks li:nth-child(1)');
+const blueBox = document.querySelector('ul.marks li:nth-child(2)');
+const greenBox = document.querySelector('ul.marks li:nth-child(3)');
+
+
+translateAllAtOnce()
+function translateAllAtOnce () { 
+  Promise.all[moveElement(redBox, {x: 20, y: 300}), moveElement(blueBox, {x: 400, y: 300}), moveElement(greenBox, {x: 400, y: 20})]
+}
+
+// translateOneByOne()
+function translateOneByOne () {
+   new Promise((resolve) => {
+      resolve(moveElement(redBox, {x: 20, y:300}))

```

Comment:
When you return a `Promise A` from `.then` function, the next `.then` will only run after the returned `A` is finished. Basically, you will be fine with just returning `moveElement(...)` from the constructor of the promise, instead of resolving it manually. I would argue, that it's a cleaner way of doing things 🙂

Not a mistake though.

Reply:
The same comment is valid in different places.